### PR TITLE
test: fix fd leaks

### PR DESCRIPTION
**Pull request by yuwata**
Fixes most issues reported in #22576.

### CODE REVIEW

**Comment by yuwata** [comment]
cc @evverx

**Comment by bluca** [comment]
```
In file included from ../src/basic/macro.h:12,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/test/test-fs-util.c:5:
In function â€˜freepâ€™,
    inlined from â€˜test_chase_symlinksâ€™ at ../src/test/test-fs-util.c:32:28:
../src/fundamental/macro-fundamental.h:293:17: error: array subscript â€˜void *[0]â€™ is partly outside array bounds of â€˜int[1]â€™ [-Werror=array-bounds]
  293 |                 free(memory);                   \
      |                 ^~~~~~~~~~~~
../src/test/test-fs-util.c: In function â€˜test_chase_symlinksâ€™:
../src/test/test-fs-util.c:32:28: note: object â€˜pfdâ€™ of size 4
   32 |         _cleanup_free_ int pfd = -1;
      |                            ^~~
In function â€˜freepâ€™,
    inlined from â€˜test_chase_symlinksâ€™ at ../src/test/test-fs-util.c:32:28:
../src/fundamental/macro-fundamental.h:293:17: error: array subscript â€˜void *[0]â€™ is partly outside array bounds of â€˜int[1]â€™ [-Werror=array-bounds]
  293 |                 free(memory);                   \
      |                 ^~~~~~~~~~~~
../src/test/test-fs-util.c: In function â€˜test_chase_symlinksâ€™:
../src/test/test-fs-util.c:32:28: note: object â€˜pfdâ€™ of size 4
   32 |         _cleanup_free_ int pfd = -1;
```

File: src/libsystemd-network/test-dhcp-client.c [hunk 1]
```diff
@@ -31,13 +31,6 @@ static bool verbose = true;
 static int test_fd[2];
 static test_callback_recv_t callback_recv;
 static be32_t xid;
-static sd_event_source *test_hangcheck;
-
-static int test_dhcp_hangcheck(sd_event_source *s, uint64_t usec, void *userdata) {
```
Comment (bluca) [inline src/libsystemd-network/test-dhcp-client.c:36]: This means we lose some test coverage, no? As this should never happen, but we don't notice anymore if a regression makes it happen?

File: src/libsystemd-network/test-dhcp6-client.c [hunk 2]
```diff
@@ -78,8 +78,6 @@ static int test_ifindex = 42;
 static unsigned test_client_sent_message_count = 0;
 static sd_dhcp6_client *client_ref = NULL;
 
-STATIC_DESTRUCTOR_REGISTER(client_ref, sd_dhcp6_client_unrefp);
```
Comment (medhefgo) [inline src/libsystemd-network/test-dhcp6-client.c:81]: This means that main() doesn't call `static_destruct()`, which it should. This whole test (sh/c)ould be converted to TEST macros. I guess I missed the stray ones that aren't located in `test/`.

File: src/libsystemd-network/test-dhcp-client.c (hunk 1 above)
Comment (yuwata) [reply src/libsystemd-network/test-dhcp-client.c:36]: No, it is replaced with `sd_event_loop() >= 0`, as if the timeout is detected it returns `-ETIMEDOUT`.

File: src/libsystemd-network/test-dhcp6-client.c (hunk 2 above)
Comment (yuwata) [reply src/libsystemd-network/test-dhcp6-client.c:81]: Yeah, converting the tests to use your macro is another choice to fix the issue. But, I'd like to leave it to later PR.

**Comment by yuwata** [comment]
Fixed the failure in test-fs-util. PTAL.

**Pull request merged**